driver/beacon_syncer.go

```diff
@@ -0,0 +1,133 @@
+package driver
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/beacon"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/taikoxyz/taiko-client/bindings"
+	"github.com/taikoxyz/taiko-client/bindings/encoding"
+	eventIterator "github.com/taikoxyz/taiko-client/pkg/chain_iterator/event_iterator"
+)
+
+// TriggerBeaconSync triggers the L2 node to start performing a beacon-sync.
+func (s *L2ChainSyncer) TriggerBeaconSync() error {
+	blockID, lastVerifiedHead, err := s.getVerifiedBlockPayload(s.ctx)
```

Comment:
```suggestion
	blockID, payload, err := s.getVerifiedBlockPayload(s.ctx)
```

The second return parameter should have a more sensible name.